tree/dataframe/src/RLoopManager.cxx

```diff
@@ -359,6 +359,66 @@ RLoopManager::RLoopManager(std::unique_ptr<RDataSource> ds, const ColumnNames_t
    fDataSource->SetNSlots(fNSlots);
 }
 
+RLoopManager::RLoopManager(ROOT::RDF::RDatasetSpec spec) // pass a hard-copy of the reference
+   : fStartEntry(spec.fStartEntry), fEndEntry(spec.fEndEntry), fNSlots(RDFInternal::GetNSlots()),
+     fLoopType(ROOT::IsImplicitMTEnabled() ? ELoopType::kROOTFilesMT : ELoopType::kROOTFiles),
+     fNewSampleNotifier(fNSlots), fSampleInfos(fNSlots)
+{
+   if (spec.fTreeNames.size() == 1) {
+      // A TChain has a global name, that is the name of single tree
+      auto chain = std::make_shared<TChain>(spec.fTreeNames[0].c_str());
+      // The global name of the chain is also the name of each tree in the list
+      // of files that make the chain.
+      for (const auto &f : spec.fFileNameGlobs)
+         chain->Add(f.c_str());
+      SetTree(chain);
+   } else {
+      // Some other times, each different file has its own tree name, we need to
+      // reconstruct the full path to the tree in each file and pass that to
+      // TChain::Add
+      auto chain = std::make_shared<TChain>();
+      for (auto i = 0u; i < spec.fFileNameGlobs.size(); i++) {
+         const auto fullpath = spec.fFileNameGlobs[i] + "?#" + spec.fTreeNames[i];
+         chain->Add(fullpath.c_str());
+      }
+      SetTree(chain);
+   }
+
+   const auto &friendNames = spec.fFriendInfo.fFriendNames;
```

Comment:
I understand the need for this. It would be very nice to have a function like `void AddFriendsToTree(TTree &tree)`, but I am yet to understand how to do this while keeping the friends alive (since they are created correctly as `unique_ptr`s they would be destroyed at the end of this hypothetic function)

Reply:
Good point. Recalling our discussion (not being able to return a unique pointer because TTree is non-movable), this might help - https://stackoverflow.com/questions/25885065/how-can-a-unique-ptr-be-returned-by-value-without-stdmove

I would also have some time to play with this once I complete the spec foundations.